command/agent/auth/auth.go

```diff
@@ -172,7 +172,7 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 			ah.logger.Debug("lookup-self with preloaded token")
 			clientToUse.SetToken(ah.token)
 
-			secret, err = clientToUse.Logical().Read("auth/token/lookup-self")
+			secret, err = clientToUse.Logical().ReadWithContext(ctx, "auth/token/lookup-self")
```

Comment:
```suggestion
			secret, err = clientToUse.Auth().Token().LookupSelfWithContext(ctx)
```

Reply:
Wondering if we should introduce our library where we can?

Reply:
Ah, good catch!